### PR TITLE
fix default value for logdateformat

### LINKED ISSUE #3107: different default values are used for parameter 'logdateformat`
refers to https://github.com/nextcloud/server/issues/3069

## Expected behavior
For all parameters default values should be set and the same default value should be used for all appearances in the source code.

## Current behavior
Not all of the parameters are set to a fixed default value, nor have they the same default value assigned in the source code.

As far as I understood the default value of the parameter `logdateformat` should be **\DateTime::ISO8601** but other values have been assigned in the source code:

```
+ getSystemValue('logdateformat', \DateTime::ISO8601)   # ok, default value !!
- /var/www/htdocs/nextcloud/apps/logreader/controller/logcontroller.php:51
- /var/www/htdocs/nextcloud/apps/logreader/controller/logcontroller.php:91

+ getSystemValue('logdateformat', 'c')                  # <<< although 'c' also represents the
                                                        # <<< ISO8601 format, the same wording
                                                        # <<< should be used.
- /var/www/htdocs/nextcloud/lib/private/Console/TimestampFormatter.php:104
```

### Steps to reproduce
1. search and compare the assigned values in the getSystemValue() call.

#### Server Configuration
OS: Linux 3.2.82
Web server: Apache2 2.4.25
Database: MariaDB 5.5.53
PHP version: 5.6.29
Nextcloud version: 11.0.0

#### Client Configuration
Browser: Mozilla Firefox 50.1.0
Operating system: Windows 7

### PR DESCRIPTION
**Pull request by MorrisJobke**
- fixes #3107 
- no functional change - only streamlines to use the very same code instead of different codes for the same value

cc @j-ed

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

File: lib/private/Console/TimestampFormatter.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function format($message) {
 		$timeZone = $timeZone !== null ? new \DateTimeZone($timeZone) : null;
 
 		$time = new \DateTime('now', $timeZone);
-		$timestampInfo = $time->format($this->config->getSystemValue('logdateformat', 'c'));
+		$timestampInfo = $time->format($this->config->getSystemValue('logdateformat', \DateTime::ISO8601));
```
Comment (icewind1991) [inline lib/private/Console/TimestampFormatter.php]: `\DateTime::ISO8601` isn't actually ISO8601 (https://secure.php.net/manual/de/class.datetime.php#datetime.constants.iso8601),  `DateTime::ATOM` is the one you're looking for

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: lib/private/Console/TimestampFormatter.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/Console/TimestampFormatter.php]: wow php, Y U not add `@deprecated` :(

Comment (MorrisJobke) [inline lib/private/Console/TimestampFormatter.php]: Sadly not what we currently log: https://secure.php.net/manual/en/function.date.php (the `c` also contains the colon) ... and I will for sure not change this format 🙈

Comment (MorrisJobke) [inline lib/private/Console/TimestampFormatter.php]: @karlitschek I'm open for changing this to proper ISO8601 - we just need to make sure we put it into the release notes.

**Comment by icewind1991** [comment]
Can we use actual ISO8601 instead of the almost ISO8601 with the notice?

**Comment by MorrisJobke** [comment]
> Can we use actual ISO8601 instead of the almost ISO8601 with the notice?

But then we break all (log management) setups that rely on the fact that our log date contains the `:` 😉

**Comment by MorrisJobke** [comment]
@icewind1991 can we get this in and decide on the log time format afterwards? Thanks :)

**Review by icewind1991 — APPROVED** [review]

**Comment by nickvergessen** [comment]
Actually you now changed the format: https://3v4l.org/MRe6f
```
var_dump(date('c'));
var_dump(date(\DateTime::ISO8601));
var_dump(date(\DateTime::ATOM));
---
string(25) "2017-01-30T15:00:45+01:00"
string(24) "2017-01-30T15:00:45+0100"
string(25) "2017-01-30T15:00:45+01:00"
```

So I will adjust this to ATOM

**Comment by MorrisJobke** [comment]
> Actually you now changed the format:

I'm now totally confused ... -.- the PHP doc says something different.

**Comment by MorrisJobke** [comment]
> I'm now totally confused ... -.- the PHP doc says something different.

Ah okay - got it.

**Comment by MorrisJobke** [comment]
> Ah okay - got it.

I compared the new format not to the new format but to itself 🙈

**Pull request merged**